src/Apps/Order/Routes/Shipping2/Components/FulfillmentDetailsForm.tsx

```diff
@@ -132,60 +150,101 @@ const FulfillmentDetailsFormLayout = (
   }
 
   const handleCloseVerification = async () => {
-    await setFieldValue("attributes.addressVerifiedBy", AddressVerifiedBy.USER)
+    await setFieldValue("meta.addressVerifiedBy", AddressVerifiedBy.USER)
     await props.onAddressVerificationComplete()
   }
 
-  const handleChooseAddress = async (verifiedBy, chosenAddress) => {
+  const serializedValues = JSON.stringify(formikContext.values)
+
+  const handleSelectSavedAddress = useCallback(
+    async (address: SavedAddressType) => {
+      shippingContext.actions.setStage("fulfillment_details")
+      await formikContext.setValues({
+        ...formikContext.values,
+        fulfillmentType: FulfillmentType.SHIP,
+        attributes: addressWithFallbackValues(address),
+        meta: {
+          ...formikContext.values.meta,
+        },
+      })
+
+      await formikContext.submitForm()
```

Comment:
After saving a user's address, the form receives it in this callback, sets all the values (because it is in saved address mode the inputs are not visible) and submits the form. FulfillmentDetails controlls submitting the values to exchange only _after_ SavedAddresses2 performs the user address updates.